web/src/containers/EncounterList/hooks.ts

```diff
@@ -24,116 +39,109 @@ export function useEncounterList(
-                    practitioner,
-                    status: encounter.status,
-                    date: encounter?.period?.start,
-                    humanReadableDate:
-                        encounter?.period?.start && formatHumanDateTime(encounter?.period?.start),
-                };
-            });
+    const patientFilterValue = filterValues?.[0]?.value ?? undefined;
+    const practitionerFilterValue = filterValues?.[1]?.value ?? undefined;
+    const dateFilterValue = filterValues?.[2]?.value ?? undefined;
+
+    const dateParameter = dateFilterValue
+        ? [
+              `ge${formatFHIRDateTime(dateFilterValue[0])}`,
+              `le${formatFHIRDateTime(dateFilterValue[1])}`,
+          ]
+        : undefined;
+
+    const queryParameters = {
+        ...searchParams,
+        _include: [
+            'Encounter:subject',
+            'Encounter:participant:PractitionerRole',
+            'PractitionerRole:practitioner:Practitioner',
+        ],
+        'participant-display': practitionerFilterValue,
+        'subject:Patient.name': patientFilterValue,
+        date: dateParameter,
+        _sort: '-_lastUpdated',
+    };
+
+    const { resourceResponse, pagerManager, handleTableChange, pagination } = usePagerExtended<
+        Encounter | PractitionerRole | Practitioner | Patient,
+        EncounterListFilterValues
+    >('Encounter', queryParameters, debouncedFilterValues);
+
+    const encounterData = mapSuccess(resourceResponse, (bundle) => {
+        const sourceMap = bundle
+            ? extractBundleResources(bundle)
+            : { Encounter: [], Patient: [], Practitioner: [], PractitionerRole: [] };
+
+        const {
+            Encounter: encounters,
+            Patient: patients,
+            Practitioner: practitioners,
+            PractitionerRole: practitionerRoles,
+        } = sourceMap;
+
+        return encounters.map((encounter) => {
+            const patient = patients.find((patient) => patient.id === encounter.subject?.id);
+
+            const practitionerRole = practitionerRoles.find(
+                (practitionerRole) =>
+                    practitionerRole.id === encounter.participant?.[0]!.individual?.id,
+            );
+
+            const practitioner = practitioners.find(
+                (practitioner) => practitioner.id === practitionerRole?.practitioner?.id,
+            );
+
+            return {
+                id: encounter.id!,
+                patient,
+                practitioner,
+                status: encounter.status,
+                date: encounter?.period?.start,
+                humanReadableDate:
+                    encounter?.period?.start && formatHumanDateTime(encounter?.period?.start),
+            };
         });
+    });
+
+    return {
+        encounterDataListRD: encounterData,
+        reloadEncounter: pagerManager.reload,
+        handleTableChange,
+        pagination,
+    };
+}
+
+export function usePagerExtended<T extends Resource, F = unknown>(
```

Comment:
I would prefer to move `usePagerExtended` from here to some other file like `src/hooks/pager.ts` or `src/components/Pager/hooks.ts` because it's now being used in many components